sor-api/src/main/java/com/bazaarvoice/emodb/sor/condition/eval/SubsetEvaluator.java

```diff
@@ -0,0 +1,485 @@
+            _left = left;
+        }
+
+        @Override
+        public final Boolean visit(ConstantCondition right, Void context) {
+            // Examples: _left subset? alwaysTrue()
+            //           _left subset? alwaysFalse()
+            if (right.getValue()) {
+                // Every condition is a subset of alwaysTrue()
+                return true;
+            }
+            // Only alwaysFalse() is a subset of alwaysFalse().  There are infinitely more contrived ones,
+            // such as and(eq("foo"),eq("bar")), but for simplicity only return true for alwaysFalse().
+            return _left instanceof ConstantCondition && !((ConstantCondition) _left).getValue();
+        }
+
+        @Override
+        public final Boolean visit(AndCondition right, Void context) {
+            // left is a subset of right only if left is a subset of each "and" condition
+            for (Condition andCondition : right.getConditions()) {
+                if (!andCondition.visit(this, context)) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        @Override
+        public final Boolean visit(OrCondition right, Void context) {
+            // left is a subset of right only if left is a subset of at least one "or" condition
+            for (Condition orCondition : right.getConditions()) {
+                if (orCondition.visit(this, context)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        @Override
+        public final Boolean visit(NotCondition right, Void context) {
+            // If possible invert the right condition
+            Condition inverseRight = InverseEvaluator.getInverseOf(right.getCondition());
+            if (inverseRight != null) {
+                return checkIsSubset(_left, inverseRight);
+            }
+
+            // If possible invert the left condition. Since "not(x) subset not(y)" == "y subset x", if the left condition
+            // can be inverted such that "not(invertedLeft)" == "left" then we can return  "right.condition subset inverseLeft"
+            Condition inverseLeft = InverseEvaluator.getInverseOf(_left);
+            if (inverseLeft != null) {
+                return checkIsSubset(right.getCondition(), inverseLeft);
+            }
+
+            // Neither left nor right have an inverse.  If the right's condition is distinct from left then it is
+            // a subset.  For example:  "ugc_us:ugc" is a subset of "not(like("*:sys"))" because "ugc_us:ugc"
+            // and "like("*:sys")" are distinct.
+
+            return DistinctEvaluator.areDistinct(_left, right.getCondition());
+        }
+
+        // All remaining methods return false be default.  Subclasses should override those methods whose return
+        // values may be true.
+
+        @Override
+        public Boolean visit(EqualCondition right, Void context) {
+            return false;
```

Comment:
similar comment as the other one - should  we throw a UnsupportedOperationException for these cases which are not implemented in the default Left Visitor? The subclasses are implementing if they are needed, otherwise its better to know I think.